pkg/match/findmatch_test.go

```diff
@@ -0,0 +1,84 @@
+package match
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_FindMatch2(t *testing.T) {
+	tt := map[string]struct {
+		lines          []int
+		needle         int
+		expectedErr    string
+		expectedValues [2]int
+	}{
+		"original example": {
+			lines:          []int{1721, 979, 366, 299, 675, 1456},
+			expectedValues: [2]int{1721, 299},
+			needle:         2020,
+		},
+		"no match": {
+			lines:       []int{1, 2, 3, 4, 5, 6},
+			expectedErr: "no match found",
+			needle:      202,
+		},
+	}
+
+	for name, tc := range tt {
+		t.Run(name, func(t *testing.T) {
+			l, m, err := FindSum2bis(tc.lines, tc.needle)
+			if tc.expectedErr != "" {
+				assert.Error(t, err)
+				assert.Equal(t, tc.expectedErr, err.Error())
+			}
+			assert.Equal(t, tc.expectedValues[0], l)
+			assert.Equal(t, tc.expectedValues[1], m)
+		})
+	}
+}
+
+func benchFindSum2_Size(lines []int, expected int, b *testing.B) {
+	b.Run("basic", func(b *testing.B) {
+		_, _, _ = FindSum2(lines, 2020)
+	})
+	b.Run("bis", func(b *testing.B) {
+		_, _, _ = FindSum2bis(lines, 2020)
+	})
+}
+
+func BenchmarkFindSum2_32(b *testing.B) {
+	lines := randomInts(32)
+	lines[16] = 1729
+	lines[30] = 299
+	for n := 0; n < b.N; n++ {
+		benchFindSum2_Size(lines, 2020, b)
+	}
+}
+
+func BenchmarkFindSum2_10k(b *testing.B) {
+	lines := randomInts(10000)
+	lines[5000] = 1729
+	lines[9899] = 299
+	for n := 0; n < b.N; n++ {
+		benchFindSum2_Size(lines, 2020, b)
+	}
+}
+
+func BenchmarkFindSum2_100k(b *testing.B) {
+	lines := randomInts(100000)
+	lines[16] = 1729
+	lines[30] = 299
+	for n := 0; n < b.N; n++ {
+		benchFindSum2_Size(lines, 2020, b)
+	}
+}
+
+func randomInts(size int) []int {
+	lines := make([]int, size)
+	for i := 0; i < size; i++ {
+		lines[i] = rand.Int()
```

Comment:
Any chance of `rand.Int()` returning couples of values that add up to 2020 ?
Just in case, add an offest here that guarantees that whatever you write in Benchmark will be the only matches.